03_The_Architecture_of_OrbitDB/05_RESTful_API.md

```diff
@@ -0,0 +1,316 @@
+## Chapter 5 - The RESTful API
+
+> Integrate OrbitDB into your applications using alternative programming languages and frameworks.
+
+<div>
+  <h3>Table of Contents</h3>
+
+- [Setting up](#setting-up)
+- [Obtaining the SSL certificates](#obtaining-the-ssl-certificates)
+- [Interacting with OrbitDB over HTTP](#interacting-with-orbitdb-over-http)
+- [Replicating](#replicating)
+- [Key takeaways](#key-takeaways)
+
+</div>
+
+### Setting up
+
+Running an OrbitDB REST server is relatively straight-forward but some knowledge of working on the command line will be required. These steps assume you are running Linux or some other Unix-based operating system. For Windows users, you will need to translate the commands to your environment.
+Prerequisites
+
+Firstly, it is assumed that you can use a command line and install software as all commands will be run from the terminal.
+
+You will also need two machines running since we will be replicating a decentralized database. This can either be two physical computers, a couple of virtual machines or docker containers.
+
+Lastly, because the OrbitDB server uses Node.js you will also need npm (bundled with Node.js) to install the dependencies. This tutorial will not cover the installation and configuration of these requirements.
+
+#### Running IPFS
+
+OrbitDB uses IPFS to distribute and replicate data stores. The OrbitDB HTTP server runs in one of two modes; local or api.
+
+When run in Local mode, OrbitDB will run its own IPFS node. When run in api mode, OrbitDB will connect to an already-running IPFS node.
+
+For this tutorial we will connect to a running IPFS daemon and will assume you already have this installed. You will also want to run IPFS daemon with pubsub enabled.
+
+Start your first IPFS daemon by running:
+
+```bash
+ipfs daemon --enable-pubsub-experiment
+```
+
+#### Building the REST server
+
+Now get a copy of the code. You can grab it via Github at https://github.com/orbitdb/orbit-db-http-api:
+
+```bash
+wget https://github.com/orbitdb/orbit-db-http-api.zip
+```
+
+Alternatively, you can clone the git repo:
+
+```bash
+git clone https://github.com/orbitdb/orbit-db-http-api.git
+```
+
+Install your dependencies:
+
+```bash
+npm install
+```
+
+### Obtaining the SSL certificates
```

Comment:
Is the SSL cert necessary to run it locally? I'm only wondering if we should move this section to the bottom to accelerate the "getting started" activity

Reply:
Technically yes, although curl can be run without ssl certs using the -k option ```curl -k ...```. @phillmac thoughts?